packages/multichain/src/caip25Permission.ts

```diff
@@ -347,3 +385,86 @@ function removeScope(
     operation: CaveatMutatorOperation.RevokePermission,
   };
 }
+
+/**
+ *
+ * @param leftValue - The existing CAIP-25 permission caveat value.
+ * @param rightValue - The incoming CAIP-25 permission caveat value.
+ * @param scopeToMerge - The required or optional scopes from the [CAIP-25](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-25.md) request.
+ * @returns The combined CAIP-25 permission caveat value.
+ */
+function mergeScopesForCaip25CaveatValue(
+  leftValue: Caip25CaveatValue,
+  rightValue: Caip25CaveatValue,
+  scopeToMerge: keyof Pick<
+    Caip25CaveatValue,
+    'optionalScopes' | 'requiredScopes'
+  >,
+): Caip25CaveatValue {
+  const newValue = cloneDeep(leftValue);
+
+  Object.entries(rightValue[scopeToMerge]).forEach(
```

Comment:
hmm... We have a `mergeScopes` helper for NormalizedScopesObject already, but I guess that's technically not usable as is here